tests/e2e/streams/pulsar_streams_tests.py

```diff
@@ -0,0 +1,383 @@
+    # should only receive message that were sent after the consumer was created. Everything
+    # inbetween should be lost. Additionally, we check that consumer continues from the correct message
+    # after stopping and starting again.
+    assert len(pulsar_topics) > 0
+    cursor = connection.cursor()
+    common.execute_and_fetch_all(
+        cursor, "CREATE PULSAR STREAM test "
+        f"TOPICS {pulsar_topics[0]} "
+        "TRANSFORM pulsar_transform.simple", )
+    common.start_stream(cursor, "test")
+    time.sleep(1)
+
+    def assert_message_not_consumed(message):
+        vertices_with_msg = common.execute_and_fetch_all(
+            cursor,
+            "MATCH (n: MESSAGE {" f"payload: '{message.decode('utf-8')}'" "}) RETURN n",
+        )
+
+        assert len(vertices_with_msg) == 0
+
+    producer = pulsar_client.create_producer(
+        common.pulsar_default_namespace_topic(
+            pulsar_topics[0]), send_timeout_millis=60000)
+    producer.send(common.SIMPLE_MSG)
+
+    check_vertex_exists_with_topic_and_payload(
+        cursor, pulsar_topics[0], common.SIMPLE_MSG)
+
+    common.stop_stream(cursor, "test")
+
+    next_message = b"NEXT"
+    producer.send(next_message)
+
+    assert_message_not_consumed(next_message)
+
+    common.start_stream(cursor, "test")
+    check_vertex_exists_with_topic_and_payload(
+        cursor, pulsar_topics[0], next_message)
+    common.stop_stream(cursor, "test")
+
+    common.drop_stream(cursor, "test")
+
+    lost_message = b"LOST"
+    valid_messages = [b"second message", b"third message"]
+
+    producer.send(lost_message)
+
+    assert_message_not_consumed(lost_message)
+
+    common.execute_and_fetch_all(
+        cursor, "CREATE PULSAR STREAM test "
+        f"TOPICS {pulsar_topics[0]} "
+        "TRANSFORM pulsar_transform.simple", )
+
+    for message in valid_messages:
+        producer.send(message)
+        assert_message_not_consumed(message)
+
+    common.start_stream(cursor, "test")
+
+    assert_message_not_consumed(lost_message)
+
+    for message in valid_messages:
+        check_vertex_exists_with_topic_and_payload(
+            cursor, pulsar_topics[0], message)
+
+
+# @pytest.mark.skip(reason="no way of currently testing this")
+@pytest.mark.parametrize("transformation", TRANSFORMATIONS_TO_CHECK)
+def test_check_stream(
+        pulsar_client,
+        pulsar_topics,
+        connection,
+        transformation):
+    assert len(pulsar_topics) > 0
+    cursor = connection.cursor()
+    common.execute_and_fetch_all(
+        cursor,
+        "CREATE PULSAR STREAM test "
+        f"TOPICS {pulsar_topics[0]} "
+        f"TRANSFORM {transformation} "
+        "BATCH_SIZE 1",
+    )
+    common.start_stream(cursor, "test")
+    time.sleep(1)
+
+    producer = pulsar_client.create_producer(
+        common.pulsar_default_namespace_topic(
+            pulsar_topics[0]), send_timeout_millis=60000)
```

Comment:
Can be the timeout be a bit smaller? I would say 10s should be plenty for this (and similar occasions), isn't it?

Reply:
I set the same thing as for Kafka (unless it's not 60 seconds but a smaller unit...).